src/cluster_asm.c

```diff
@@ -678,20 +682,17 @@ void clusterSyncSlotsSnapshotEOF(client *c) {
     task->rdb_channel_state = ASM_RDBCHANNEL_DONE;
     serverLog(LL_NOTICE,
         "RDB channel snapshot transfer done for task");
-
-    task->state = ASM_REPLAY_STREAM;
-
-    /* Replay stream. */
-    asmSyncBufferStreamToDb(task);
-    serverLog(LL_NOTICE, "Replaying stream for task is done");
-    connSetReadHandler(task->main_channel_conn, readQueryFromClient);
-
-    /* ACK offset during replaying buffer stream, fake offset. */
-    sendCommand(task->main_channel_conn, "CLUSTER", "SYNCSLOTS", "ACK", "123", NULL);
     /* Free the RDB channel connection. */
     c->task = NULL;
     c->flags &= ~CLIENT_MASTER;
     freeClientAsync(c); /* Free the client, it is no longer needed. */
+    task->rdb_channel_conn = NULL;
+
+    /* Will start streaming the buffer to DB, don't start here since now
+     * we are in the context of executing command, otherwise, redis will
+     * generate a big MULTI-EXEC including all the commands in the buffer.
+     * just update the state here, and do it in beforeSleep(). */
```

Comment:
hmm, good point. For replicaof, we were doing it inside connection handler. Now, here we can't do it in the command handler.

Reply:
yes, thanks for confirm